File: capstone/java/src/main/java/com/techelevator/service/RecipeService.java
package com.techelevator.service;

import com.techelevator.model.Recipe;

import java.security.Principal;
import java.util.List;

public interface RecipeService {
    List<Recipe> getRecipes(Principal user);
    Recipe getRecipeById(int id, Principal user);
    Recipe createRecipe(Recipe recipe, Principal user);
    Recipe updateRecipeInfo(int id, Recipe recipe, Principal user);
    int addIngredientToRecipe(int recipeId, int ingId, int msmId, double quantity, Principal user);
    int removeIngredientFromRecipe(int recipeId, int ingId, Principal user);
}

File: capstone/vue/src/services/RecipeService.js
import axios from 'axios';

export default{
    getRecipes(){
        return axios.get('/recipes')
    },

    getRecipeById(id){
        return axios.get(`/recipes/${id}`)
    },

    addRecipe(recipe){
        return axios.post('/recipes', recipe)
    },

    updateRecipe(recipe){
        return axios.put(`/recipes/${recipe.id}`, recipe)
    },

    deleteRecipe(id){
        return axios.delete`/recipes/${id}`
    },

    addIngredientToRecipe(recipeId, ingredientId, msmId, quantity){
        return axios.post(`/recipes/${recipeId}/modify/${ingredientId}`, null,
        {params: {msmId, quantity}})
    }
}